lib/topologies/server.js

```diff
@@ -247,8 +249,11 @@ var eventHandler = function(self, event) {
       // Query options
       var queryOptions = { numberToSkip: 0, numberToReturn: -1, checkKeys: false, slaveOk: true };
       // Create a query instance
-      var query = new Query(self.s.bson, 'admin.$cmd', {ismaster:true, client: self.clientInfo}, queryOptions);
-      // Get start time
+      if (self.s.compression && self.s.compression.compressors) {
```

Comment:
it looks like this all could be reduced to:
```
var compressors = (self.s.compression && self.s.compression.compressors) ? self.s.compression.compressors : [];
 var query = new Query(self.s.bson, 'admin.$cmd', {ismaster:true, client: self.clientInfo, compression: compression}, queryOptions);
```

Also, its curious that you seem to be assigning `compressors` to a property called `compression`, and not `self.s.compression`. Is that intentional?